ee/app/models/server/models/LivechatRooms.js

```diff
@@ -41,6 +41,7 @@ LivechatRooms.prototype.findAbandonedOpenRooms = function(date) {
 	return this.find({
 		'omnichannel.predictedVisitorAbandonmentAt': { $lte: date },
 		waitingResponse: { $exists: false },
+		servedBy: { $exists: true },
```

Comment:
Even though this solution might work, I think there's a better way to fix the issue.
I can't see any reason to keep `omnichannel.predictedVisitorAbandonmentAt` stored in the document, so I'd suggest something like:

- Create a new callback to be trigger after the chat returns to the queue;
- On `EE` folder (hooks), listen to the new hook/callback;
- Check if the room has the `predictedVisitorAbandonmentAt` property
- [Remove](https://github.com/RocketChat/Rocket.Chat/blob/develop/ee/app/models/server/models/LivechatRooms.js#L74) the field from the document

Reply:
Thanks for the instructions. It really helped quickly solve this